### ISSUE TITLE
Run time error 'System.AccessViolationException' when trying to run the Face detection example. 

### ISSUE THREAD

**Issue by AndrewKaplanovsky** [cureos/accord-samples#1]
Hi, 

When I try to run the face detection example, I get this error:

An unhandled exception of type 'System.AccessViolationException' occurred in Xamarin.Forms.Platform.WinRT.DLL

Additional information: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.

I tried to run the following solutions:
- FaceDetection.WinPhone81
- FaceDetection.Windows

Both raise the same error.

Project FaceDetection, MainPage.xaml.cs,
private void OnDetectButtonClicked(object sender, EventArgs args):

  if (objects.Length > 0)
  {
         var marker = new RectanglesMarker(objects, Color.FromArgb(0xff, 0xff, 0x00, 0xff));

```
     var stream = new MemoryStream();
     marker.Apply(this.bitmap).Save(stream, ImageFormat.Jpeg); <---- Here I check the stream in debug mode, and it is empty after saving the image into it

     this.ImageView.Source = this.GetImageSourceFromStream(stream);
```

  }

Regards,
Andrew Kaplanovsky

**Comment by anders9ustafsson** [comment]
@AndrewKaplanovsky Which versions of the Accord, Shim and Shim.Drawing libraries are you using? It's been a while since I updated the references in these projects, this might play a role.

I will look closer into this issue when time permits.

**Comment by AndrewKaplanovsky** [comment]
Hi Anders,

In the app I'm developing:
Accord 3.0.2.60302
Shim 2.2.0
Shim.Drawing 3.0.0

In accord-samples that I downloaded from GH, FaceDetection:
Accord 3.0.2.50804
Shim 1.0.2.50909
Shim.Drawing 2.2.5.50526

In both cases I get the same error:

When I create a new Stream, I see that its length is zero

![1 - new stream](https://cloud.githubusercontent.com/assets/2852069/16121904/f5637f02-33dc-11e6-8409-8f40eccda438.png)

After the Bitmap  is saved into the stream, its length is still zero

![2 - after the bitmaps was saved to the memory stream](https://cloud.githubusercontent.com/assets/2852069/16121910/f9728052-33dc-11e6-9630-1ca2a9f0b1bb.png)

After that the app fails with the 'System.AccessViolationException' trying to display the processed image.

Thank you,
Andrew

**Comment by anders9ustafsson** [comment]
OK, I have looked a little closer into this issue, and it seems like there is a problem with `Bitmap.Save` for the Universal (8.1) platform. I will attempt a bug fix as soon as possible. In the mean time, many thanks for reporting, @AndrewKaplanovsky !

**Comment by AndrewKaplanovsky** [comment]
@anders9ustafsson, thank you so much for your help! I'll wait for the fix.

**Comment by anders9ustafsson** [comment]
@AndrewKaplanovsky I have implemented a fix and verified that it works. In a short while I will publish a new _Shim.Drawing_  release with the bug fix included.

**Comment by AndrewKaplanovsky** [comment]
@anders9ustafsson, thank you very much!

**Comment by AndrewKaplanovsky** [comment]
@anders9ustafsson Hi Anders, sorry for disturbing you, but when it is possible to expect the fixed version of the Shim.Drawing dll for the Universal platform published?

**Comment by anders9ustafsson** [comment]
@AndrewKaplanovsky I am sorry about the delay, quite a few things have gotten in the way. I'll see what I can do within the next couple of days, but I am afraid I cannot promise anything for sure.

**Comment by anders9ustafsson** [comment]
@AndrewKaplanovsky I have now published version 3.0.1 of _Shim.Drawing_ on NuGet, with the fix for `Image.Save` on Windows Universal. Please give it a try and see if it works sufficiently for you.

**Comment by AndrewKaplanovsky** [comment]
Anders, thank you very much, it works now!
